### DISCUSSION TITLE
Mask right curly braces

### DISCUSSION THREAD

**Discussion by cbot59** [logfellow/logstash-logback-encoder#1009] [Q&A] [answered]
How do I put <mask> value of such regex? `(password)=(.*?(, |$|}]\"))?` (java regex, double quote is escaped)

Context: 
I'm using `MaskingJsonGeneratorDecorator` and unable to run the app with above regex.
Here's my current setup

```
<jsonGeneratorDecorator class="net.logstash.logback.mask.MaskingJsonGeneratorDecorator">
                    <valueMask>
                        <value>(password)=(.*?(, |$|}]"))?</value>
                        <mask>$1=[filtered]$3</mask>
                    </valueMask>
</jsonGeneratorDecorator>
```

When I debug, the `RegexValueMasker` stores the `Pattern` as `(.*?(, |$|` which is an invalid regex hence lead to an error.
Am I missing something or is it a bug? I tried to escape the `}` using `\`, `\\`, `&#125;` but none works

Here's my sample input `taxPayerPassportNumber=someone@gmail.com}]","thread"`
I expect it will be masked into `taxPayerPassportNumber=[filtered]}]","thread"`

**Comment by philsttr** [comment, marked as answer]
Seems to be a bug in logback.  Filed https://github.com/qos-ch/logback/issues/836
